rollup/ui5-resolve.js

```diff
@@ -0,0 +1,118 @@
+
+    return {
+
+        name: "ui5-resolve",
+
+        buildStart: async (buildOptions) => {
+            const project = await normalizer.generateProjectTree({
+                cwd: options.projectPath
+            });
+            this.dependencies = resourceFactory.createCollectionsForTree(project, {}).dependencies;
+
+            const resources = await this.dependencies.byGlob("/resources/sap/ui/fl/**");
+            const writePromises = resources.map(resource => {
+                const file = `./dist${resource.getPath()}`;
+                const folder = path.dirname(file);
+                if (!fs.existsSync(folder)) {
+                    fs.mkdirSync(folder, { recursive: true });
+                }
+                if (fs.existsSync(file)) {
+                    fs.unlinkSync(file);
+                }
+                return pipe(resource.getStream(), fs.createWriteStream(file));
+            });
+            await Promise.all(writePromises);
+        },
+
+
+        // Right before writting result to dist
+        renderChunk: async (code) => {
+            return "global.window = {};" + code;
+        },
+
+
+        buildEnd: async (error) => {
+            log.verbose("buildEnd", error);
+        },
+
+
+        resolveId: (source, importer) => {
+            log.verbose(`resolveId: ${source} from ${importer}`);
+            if (importer && source.startsWith(".")) {
+                source = path.posix.join(path.dirname(importer), source);
+            }
+            log.verbose(" --> resolve to: " + source);
+            return source;
+        },
+
+
+        load: async (id) => {
+            log.verbose(`load: ${id}`);
+
+            const localFile = path.join(process.cwd(), id);
+            if (fs.existsSync(localFile)) {
+                log.info(`Using local file "${localFile}"`);
+                return fs.readFileSync(localFile, {
+                    encoding: "utf8"
+                });
+            }
+
+            const localOverride = path.resolve(__dirname, "overrides", id + ".js");
+            if (fs.existsSync(localOverride)) {
+                log.info(`Using local override for "${id}"`);
+                return fs.readFileSync(localOverride, {
+                    encoding: "utf8"
+                });
+            }
+
+            const resource = await this.dependencies.byPath(`/resources/${id}.js`);
+            return resource.getString();
+        },
+
+
+        transform: (code, id) => {
+            const skipped = !filter(id);
+            log.verbose(`transform: ${id} ${skipped ? "skipped" : ""}`);
+            if (skipped) {
+                return;
+            }
+            code = code
+                .replace(/sap\.ui\.define/g, "define")
+                .replace(/\, \/\* bExport\= \*\/ true\)/g, ")")
+                .replace(/}, true\);$/g, "});");
+
+            if (code) {
+                try {
+                    const result = amdextract.parse(code);
```

Comment:
I think the unused import check is not needed for your scenario

you can also get rid of the amdextract dependency IMO

the complete try catch can be removed!

Reply:
Less code is better ;)